frontend/src/app/(pages)/questions/api.tsx

```diff
@@ -0,0 +1,112 @@
+import api from "@/helpers/endpoint";
```

Comment:
Is this `api.tsx` file supposed to contain only the functions for API calls? If so, do you consider moving this file to the `api` directory instead of `(pages)`? And also you can make use of [NextJS route handler](https://nextjs.org/docs/app/building-your-application/routing/route-handlers) to handle the HTTP methods. Ignore this suggestion if this `api.tsx` file is for temporary testing only.

Reply:
yes, but these methods are only for question pages to consume. if i'm not wrong route handler pattern is to expose the http methods for public access? essentially like a backend api server. unless there's a way to block external access?

Reply:
Hmm, I didn't try blocking external calls to NextJS API before. But I am guessing there should be a way to do that. And eventually, we are using a centralized API in our own network to communicate with all the services right? So I think even if the API is publicly accessible, it is fine. I will let you decide whether you want to move the code.

Reply:
I dont think it is a good idea to expose api from the frontend, it should only consume api from the backend services.
The methods in api.tsx are just wrapper functions of api() in the endpoint.ts that can be shared across services. so I guess there is a centralized layer for consuming external apis at the same time there's some flexibility for the upper layers to declare their own wrapper functions.

Reply:
I believe in that case, you should move your `api.tsx` code to `utils` or `lib` folder instead of `(pages)`. And also rename it as something else (eg: `/utils/question/wrapper.ts` to separate helper functions, API, and frontend code.